src/Extensions/TimedScopes/TimedScope.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.Logging.TimedScopes
+{
+	/// <summary>Logs duration of activity</summary>
+	public class TimedScope : IDisposable
+	{
+		/// <summary>TimedScope result</summary>
+		public TimedScopeResult Result { get; set; }
+
+
+		/// <summary>TimedScope sub tipe</summary>
+		public string SubType { get; set; }
+
+
+		/// <summary>TimedScope meta data</summary>
+		public string MetaData { get; set; }
+
+
+		internal TimedScope(TimedScopeEventSource eventSource, Activity activity, string serviceName, TimedScopeResult result, ILogReplayer? logReplayer)
+		{
+			m_eventSource = eventSource;
+			m_logReplayer = logReplayer;
+			m_serviceName = serviceName;
+			m_activity = activity;
+			Result = result;
+			SubType = NullPlaceholder;
+			MetaData = NullPlaceholder;
+		}
+
+
+		/// <summary>Start TimedScope</summary>
+		public TimedScope Start()
+		{
+			m_activity.Start();
+			return this;
+		}
+
+
+		/// <summary>Stop TimedScope and log informations about it</summary>
+		public void Stop()
+		{
+			m_activity.Stop();
+			m_eventSource.LogEvent(
+				name: m_activity.OperationName,
+				subtype: SubType,
+				metadata: MetaData,
+				userHash: m_activity.GetUserHash(), //Breaking Change: feild not set
+				serviceName: m_serviceName,
+				result: Result,
+				correlationId: m_activity.Id,
+				durationMs: m_activity.Duration.TotalMilliseconds,
+				isTransaction: m_activity.IsTransaction()); //Breaking Change: feild not set
+
+			if (m_logReplayer != null && ShouldReplayEvents)
+			{
+				m_logReplayer.ReplayLogs(m_activity);
+			}
+		}
+
+
+		void IDisposable.Dispose() => Stop();
```

Comment:
>Dispose [](start = 19, length = 7)

This isn't the optimal pattern for an `IDisposable` class. Please see [https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose) for the best method

Reply:
In this case, it's the goal of Dispose is not to deallocate resources but only to provide convenience syntax of using TimedScopes in `using` block, so I don't want to add anything that will delay deletion of the object, but I've added a flag from the pattern to avoid any actions if it was called second time